### ISSUE TITLE
Mobile device - gesture top-down over carousel not working properly.

### ISSUE THREAD

**Issue by dalibor-sanzeru** [Ks89/angular-modal-gallery#186]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Feature request
[ ] Documentation issue or request
[X ] Support request
</code></pre>

## Current behavior

On carousel on Mobile device finger gestures are not working as expected.

## Expected behavior

When carousel used on cell phone, finger gesture from top-down and down-top over carousel should scroll the gallery down or up.

## Minimal reproduction of the problem with instructions
- open provided link (https://ks89.github.io/angular-modal-gallery-2018-v7.github.io/demo/carousel) on mobile phone and try finger gesture top-down over image in carousel.

## What is the motivation / use case for changing the behavior?

- user wants to move gallery images in carousel LEFT<->RIGHT not TOP<->DOWN (this gesture should move page itself).

## Environment (the most important section to fill very carefully)

<pre><code>
CURRENT GALLERY VERSION EXAMPLES WEB PAGE VERSION:
https://ks89.github.io/angular-modal-gallery-2018-v7.github.io/demo/carousel

- @ks89/angular-modal-gallery version: X.X.X
- Node version: X.X.X  
- npm version: X.X.X  
- Operating System and version:  
- Angular version: X.Y.Z 
- angular-cli version (or SystemJS/Webpack): X.Y.Z 
- I'm using Server Side Rendering with angular-universal: YES/NO
- I'm compiling with mode: DEBUG / PROD / PROD with AOT

Browser:
- [X ] Chrome (desktop) version XX
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX

Others:

</code></pre>

**Comment by Ks89** [comment]
Thanks for the issue. I'll investigate in a couple of days.

**Comment by Ks89** [comment]
Which mobile device are you using? And which os version?

Did you try also with my official documentation website https://ks89.github.io/angular-modal-gallery-2018-v7.github.io/demo/carousel-fixed ?

**Comment by dalibor-sanzeru** [comment]
Hi, 
Sorry for late response.

This problem was confirmed and tested on official documentation page which you sent. 
- tested on devices: 
 - ZenFone 5 (ZE620KL) - Oreo OS.
- lenovo Z2 Vibe - Android 5.0 Lollipop
- Chrome DEv tools simulated with same behaiour - I am pasting image for better imagination.
![image](https://user-images.githubusercontent.com/19836016/55667988-a2eda280-5863-11e9-800d-045bfcdc5c0e.png)

**Comment by Ks89** [comment]
Ohh ok, now I understood the issue.
Yes I know it, this happens because I'm using 'hammerjs' to handle swipe, as suggested by Angular team.
I can't do anything to change this behaviour.
Some months ago I openend an issue on hammerjs github page, but I didn't receive an answer.
It seems that the hammerjs community is not so active as it should be. The are more than 200 open issues without solution.

If you have a working alternative to handle swipe effects easily, feel free to share it.

I hope that Angular team will add an official angular's submodule to handle swipes in the future.

**Comment by mohaxspb** [comment]
@Ks89 Hello!

Is there are some news for this isuue? May be there is a way to disable swipes on mobile through library attributes?

**Comment by mohaxspb** [comment]
I found a temporary solution [here](https://stackoverflow.com/a/42425301/3212712) with disabling hammer for concrete swipe action.

Just need to modify `app.module.ts` so:

```
import {HAMMER_GESTURE_CONFIG, HammerGestureConfig} from '@angular/platform-browser';

export class MyHammerConfig extends HammerGestureConfig {
  buildHammer(element: HTMLElement) {
    return new Hammer(element, {
      touchAction: 'pan-y',
    });
  }
}


@NgModule({
  providers: [
    { provide: HAMMER_GESTURE_CONFIG, useClass: MyHammerConfig }
  ]
})
export class AppModule {
}
```

**Comment by Ks89** [comment]
@mohaxspb mmm interesting and if you change the library in this way, can you still use horizontal swipes without issues?

If yes would you create a pull request with your solution on master branch?

**Comment by mohaxspb** [comment]
I add this code to my project and yes, this allows to use horizontal swipes.

I tryed to test if it'll work, if place this code in library, but can't find a way to run example with modified lib. 

Maybe I'm doing something wrong, as I'm new to angular and this is the first library, that I use)

Here is the error in console, that if found after adding code to library, building it and installing for example:

```
core.js:14597 ERROR Error: Uncaught (in promise): Error: StaticInjectorError(AppModule)[ModalGalleryComponent -> InjectionToken Platform ID]: 
  StaticInjectorError(Platform: core)[ModalGalleryComponent -> InjectionToken Platform ID]: 
    NullInjectorError: No provider for InjectionToken Platform ID!
Error: StaticInjectorError(AppModule)[ModalGalleryComponent -> InjectionToken Platform ID]: 
  StaticInjectorError(Platform: core)[ModalGalleryComponent -> InjectionToken Platform ID]: 
    NullInjectorError: No provider for InjectionToken Platform ID!
    at 
```

**Comment by Ks89** [comment]
I'm sorry for the delay.

If you want to build the library please check the CONTRIBUTING.MD file in the repo. 

Your changes must be applied to the lib folder and not to the main example. If you need more info feel free to ask.

**Comment by Ks89** [comment]
@mohaxspb If you need additional support feel free to ask.

**Comment by mohaxspb** [comment]
@Ks89 sorry for delay too)

Seems to be I managed to correctly build. See https://github.com/Ks89/angular-modal-gallery/pull/187 with fix. 

As it's my first PR for angular, npm and so on I can made something wrong, but it really fixes issue)
Also I have no idea about what this can affects, so, if I'll merge it to master, I think you should make a note in README about it to prevent unexpected behavior after update)

**Comment by Ks89** [comment]
@mohaxspb i merged the PR and released as 7.2.2.
It seems ok and backward compatible. Good catch and thank u for the contribution.

Feel free to create other PR, but if you want to do big features it should be better if we talk about the idea before to start to work.

👍

**Comment by mohaxspb** [comment]
@Ks89 great! OK, thanks)
